app/native/IRContextMenuManager/IRContextMenuManager.mm

```diff
@@ -0,0 +1,174 @@
+
+RCT_EXPORT_MODULE()
+
+- (std::shared_ptr<facebook::react::TurboModule>)getTurboModule:(const facebook::react::ObjCTurboModule::InitParams &)params {
+  return std::make_shared<facebook::react::NativeIRContextMenuManagerSpecJSI>(params);
+}
+
+#pragma mark - API
+
+- (void)showContextMenu:(NSArray *)items {
+  RCTExecuteOnMainQueue(^{
+    [self presentContextMenuWithItems:items];
+  });
+}
+
+#pragma mark - Helpers
+
+- (void)presentContextMenuWithItems:(NSArray *)items {
+  NSMenu *menu = [[NSMenu alloc] initWithTitle:@""];
+
+  [self buildMenu:menu fromEntries:items path:@[]];
+
+  NSEvent *event = [NSEvent mouseEventWithType:NSEventTypeRightMouseDown
+                                       location:[self currentMouseLocation]
+                                  modifierFlags:0
+                                      timestamp:[[NSProcessInfo processInfo] systemUptime]
+                                   windowNumber:[[NSApp keyWindow] windowNumber]
+                                        context:nil
+                                    eventNumber:0
+                                     clickCount:1
+                                       pressure:1.0];
+
+  [NSMenu popUpContextMenu:menu withEvent:event forView:[[NSApp keyWindow] contentView]];
+}
+
+- (NSPoint)currentMouseLocation {
+  NSPoint screenPoint = [NSEvent mouseLocation];
+  
+  // Convert screen -> window coordinates for the key window
+  NSWindow *window = [NSApp keyWindow];
+  if (!window) {
+    return screenPoint; // best effort
+  }
+  
+  NSRect screenRect = NSMakeRect(screenPoint.x, screenPoint.y, 0, 0);
+  NSPoint windowPoint = [window convertRectFromScreen:screenRect].origin;
+  return windowPoint;
+}
+
+- (void)buildMenu:(NSMenu *)menu fromEntries:(NSArray *)entries path:(NSArray<NSString *> *)path {
+  for (id entry in entries) {
+    if ([entry isKindOfClass:[NSString class]] && [(NSString *)entry isEqualToString:kSeparatorString]) {
+      [menu addItem:[NSMenuItem separatorItem]];
+      continue;
+    }
+
+    if (![entry isKindOfClass:[NSDictionary class]]) continue;
+    NSDictionary *item = (NSDictionary *)entry;
+    NSString *label = item[@"label"] ?: @"";
+    BOOL enabled = item[@"enabled"] != nil ? [item[@"enabled"] boolValue] : YES;
+    BOOL checked = item[@"checked"] != nil ? [item[@"checked"] boolValue] : NO;
+    NSString *shortcut = item[@"shortcut"] ?: @"";
+    NSArray *children = item[@"children"];
+
+    NSMenuItem *menuItem = [[NSMenuItem alloc] initWithTitle:label action:enabled ? @selector(_ir_menuItemPressed:) : nil keyEquivalent:@""];
+    menuItem.target = enabled ? self : nil;
+    menuItem.enabled = enabled;
+    menuItem.state = checked ? NSControlStateValueOn : NSControlStateValueOff;
+    if (shortcut.length > 0) {
+      [self applyShortcut:shortcut toItem:menuItem];
+    }
+
+    NSArray<NSString *> *currentPath = ({
+      NSMutableArray *m = [path mutableCopy];
+      if (label) [m addObject:label];
+      [m copy];
+    });
```

Comment:
Avoid using this sort of statement expression, which isn't common in obj-c IMO.

A better way might be a ternary:

```suggestion
  NSArray<NSString *> *currentPath = label ? [path arrayByAddingObject:label] : [path copy];
```